experimental/packages/opentelemetry-sdk-metrics-base/src/export/PeriodicExportingMetricReader.ts

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { MetricReader, promiseWithTimeout } from './MetricReader';
+import { MetricExporter } from './MetricExporter';
+import { MetricData } from './MetricData';
+import { ReaderResult, ReaderResultCode } from './ReaderResult';
+
+export type PeriodicExportingMetricReaderOptions = {
+  exporter: MetricExporter
+  exportIntervalMillis?: number,
+  exportTimeoutMillis?: number
+}
+
+/**
+ * {@link MetricReader} which collects metrics based on a user-configurable time interval, and passes the metrics to
+ * the configured {@link MetricExporter}
+ */
+export class PeriodicExportingMetricReader extends MetricReader {
+  private _interval?: ReturnType<typeof setInterval>;
+
+  private _exporter: MetricExporter;
+
+  private readonly _exportInterval: number;
+
+  private readonly _exportTimeout: number;
+
+  constructor(options: PeriodicExportingMetricReaderOptions) {
+    super(options.exporter.getPreferredAggregationTemporality());
+
+    if (options.exportIntervalMillis !== undefined && options.exportIntervalMillis <= 0) {
+      throw Error('exportIntervalMillis must be greater than 0');
+    }
+
+    if (options.exportTimeoutMillis !== undefined && options.exportTimeoutMillis <= 0) {
+      throw Error('exportTimeoutMillis must be greater than 0');
+    }
+
+    if (options.exportTimeoutMillis !== undefined &&
+      options.exportIntervalMillis !== undefined &&
+      options.exportIntervalMillis < options.exportTimeoutMillis) {
+      throw Error('exportIntervalMillis must be greater than or equal to exportTimeoutMillis');
+    }
+
+    this._exportInterval = options.exportIntervalMillis ?? 60000;
+    this._exportTimeout = options.exportTimeoutMillis ?? 30000;
+    this._exporter = options.exporter;
+  }
+
+  private async _runOnce(): Promise<void> {
+    const collectionResult = await new Promise<ReaderResult<MetricData[]>>(resolve =>
+      this.collect({ done: resolve }));
+
+    // Throw on any code that does not indicate success.
+    if (collectionResult.code !== ReaderResultCode.SUCCESS) {
+      throw collectionResult.error ?? new Error('Unknown error occurred during collection.');
+    }
+
+    // This case should never occur.
+    if (collectionResult.returnValue == null) {
+      throw new Error('Unknown error occurred during collection.');
+    }
+
+    await this._exporter.export(collectionResult.returnValue);
+  }
+
+  protected onInitialized(): void {
+    // start running the interval as soon as this reader is initialized and keep handle for shutdown.
+    this._interval = setInterval(async () => {
+      promiseWithTimeout(this._runOnce(), this._exportTimeout, (result => {
```

Comment:
Should collection duration be counted in the export timeout?

Reply:
It seems like it is handled not at all or including collect in the Java and .NET SDK implementations. According to open-telemetry/opentelemetry-specification#2140 the part of the part the describes`PeriodicExportingMetricReader` has been left intentionally vague, I think this also applies for this.